junit-jupiter-engine/src/test/kotlin/org/junit/jupiter/api/KotlinAssertionsTests.kt

```diff
@@ -60,6 +61,14 @@ class KotlinAssertionsTests {
         assertThrows<AssertionError>({ "should fail" }) { fail(null as Throwable?) }
     }
 
+    @Test
+    fun `expected context exception testing`() = runBlocking<Unit> {
+        assertThrows<AssertionError>("Should fail async") { failAsync("Should fail async") }
```

Comment:
Maybe you can use an anonymous function here to avoid having to create the private function below, which clutters up the tests:

```
        assertThrows<AssertionError>("Should fail async") {
            suspend { fail("Should fail async") }()
        }
```

Reply:
Agree, creating coroutine in place looks good